src/engine/ruleset/decoders/fortinet/fortinet-firewall.yml

```diff
@@ -0,0 +1,505 @@
+    map:
+      - destination.ip: $~tmp.remip
+  - check:
+    - ~tmp.locip: +ef_exists
+    - ~tmp.srcip: +ef_not_exists
+    map:
+      - source.ip: $~tmp.locip
+  - map:
+    - related.ip: [$source.ip]
+  - check:
+    - log.level: error
+    map:
+      - event.outcome: failure
+  - check:
+    - source.ip: +ef_exists
+    map:
+      - ~tmp.ip: +s_ip_version/$source.ip
+  - check:
+    - destination.ip: +ef_exists
+    - source.ip: +ef_not_exists
+    map:
+      - ~tmp.ip: +s_ip_version/$destination.ip
+  - map:
+    - network.type: +s_lo/$~tmp.ip
+
+  # Check-mapping for firewall-event
+  - check:
+    - ~tmp.subtype: user
+    - event.action: +r_match/(?i)logon
+    map:
+      - event.type: [start, user]
+  - check:
+    - ~tmp.subtype: user
+    - event.action: +r_match/(?i)logoff
+    map:
+      - event.type: [end, user]
+  - map:
+    - event.category: [authentication]
+  - check:
+    - ~tmp.subtype: system
+    map:
+      - event.category: [host]
+      - event.type: [info]
+      - event.outcome: +ef_delete
+
+  # Check-mapping for firewall-traffic
+  - check:
+    - ~tmp.subtype: vpn
+    map:
+      - event.category: [network]
+      - event.type: [connection]
+  - check: +s_eq/~tmp.type/traffic OR +s_eq/~tmp.type/utm
+    map:
+      - event.category: [network]
+  - check:
+    - ~tmp.type: traffic
+    map:
+      - event.type: [connection]
+  - check: >-
+        +s_eq/~tmp.type/traffic AND (+s_eq/event.action/block OR +s_eq/event.action/blocked OR
+        +s_eq/event.action/deny OR +s_eq/event.action/close OR
+        +s_eq/event.action/server-rst OR +s_eq/~tmp.utmaction/block)
+    map:
+      - event.type: +a_append/denied
+  - check: >-
+        +s_eq/~tmp.type/traffic AND +ef_not_exists/~tmp.utmaction AND
+        (+s_eq/event.action/dns OR +s_eq/event.action/accept OR +s_eq/event.action/ip-conn)
+    map:
+      - event.type: +a_append/allowed
+  - check:
+    - ~tmp.type: traffic
+    - event.action: allow
+    map:
+      - event.type: +a_append/allowed
+  - check:
+    - ~tmp.type: traffic
+    - event.action: +s_eq/start
+    map:
+      - event.type: +a_append/start
+  - check:
+    - ~tmp.type: traffic
+    - event.action: +s_ne/start
+    map:
+      - event.type: +a_append/end
+  - check:
+    - ~tmp.type: traffic
+    - event.action: +s_ne/deny
+    - network.application: +ef_exists
+    map:
+      - event.type: +a_append/protocol
+  - check:
+    - ~tmp.type: traffic
+    - network.packets: +ef_not_exists
+    - source.packets: +ef_exists
+    map:
+    - network.packets: $source.packets
+  - check: +s_eq/~tmp.type/traffic AND +ef_exists/destination.ip
+    map:
+      - related.ip: +a_append/$destination.ip
+
+  # Check-mapping for firewall-utm
+  - check: >-
+        +s_eq/~tmp.type/utm AND (+s_eq/event.action/block OR
+        +s_eq/event.action/blocked OR +s_eq/event.action/deny OR
+        +s_eq/event.action/close OR +s_eq/event.action/server-rst OR
+        +s_eq/event.action/dropped)
```

Comment:
Is it possibly faster to test whether `event.action` is not of certain types (inequality)?